pkgs/by-name/ai/aider-chat/package.nix

```diff
@@ -0,0 +1,109 @@
+{
+  lib,
+  stdenv,
+  python311,
+  fetchFromGitHub,
+  git,
+  portaudio,
+}:
+
+let
+  python3 = python311.override {
+    self = python3;
+    packageOverrides = _: super: { tree-sitter = super.tree-sitter_0_21; };
+  };
+in
+python3.pkgs.buildPythonApplication rec {
```

Comment:
I don't think using `rec` here is justified. Just moved the `version` attribute in the `let..in` clause, then `inherit` it in the main body, and you'll be able to remove `rec`.